zipline/assets/futures.py

```diff
@@ -126,15 +126,13 @@ def _maybe_update_current_chain(self):
             list
                 The up-to-date current chain, a list of Future objects.
         """
-        dt = self._get_datetime()
-
-        if (self._last_updated is None) or (self._last_updated != dt):
+        if (self._last_updated is None)\
+                or (self._last_updated != self.as_of_date):
```

Comment:
For a given FutureChain will the `as_of_date` ever change? If not then the `if (self._last_updated is None) or (self._last_updated != self.as_of_date):` clause will always be `True`, so we can get rid of it.